store/postgres/src/store.rs

```diff
@@ -257,15 +312,36 @@ impl StoreTrait for Store {
         self.schema_provider_event_sink.clone()
     }
 
-    fn event_stream(&mut self) -> Result<Receiver<StoreEvent>, StreamError> {
-        // If possible, create a new channel for streaming store events
-        match self.event_sink {
-            Some(_) => Err(StreamError::AlreadyCreated),
-            None => {
-                let (sink, stream) = channel(100);
-                self.event_sink = Some(sink);
-                Ok(stream)
-            }
+    fn subscribe(
+        &mut self,
+        subgraph: String,
+        entities: Vec<String>,
+    ) -> (String, Box<Stream<Item = EntityChange, Error = ()> + Send>) {
+        // Prepare the new subscription by creating a channel and a subscription object
+        let (sender, receiver) = channel(100);
+        let id = Uuid::new_v4().to_string();
+        let subscription = Subscription {
+            subgraph,
+            entities,
+            sender,
+        };
+
+        // Add the new subscription (log an error if somehow the UUID is not unique)
+        let subscriptions = self.subscriptions.clone();
+        let mut subscriptions = subscriptions.write().unwrap();
+        if subscriptions.contains_key(&id) {
+            error!(self.logger, "Duplicate Store subscription detected; \
+                                 subscriptions will not work as expected";
+                   "id" => &id);
         }
+        subscriptions.insert(id.clone(), subscription);
+
+        // Return the subscription ID and entity change stream
+        (id, Box::new(receiver))
+    }
+
+    fn unsubscribe(&mut self, id: String) {
+        let subscriptions = self.subscriptions.clone();
+        subscriptions.write().unwrap().remove(&id);
```

Comment:
Is this sophisticated cancellation system necessary? Wouldn't it be sufficient to drop the receiver, then if a `send` fails in `handle_entity_changes` that sender is removed from the list?

Reply:
I'm ok doing it that way. It's definitely cleaner than this extra logic.

Reply:
Also: sophisticated? 🤣